### ISSUE TITLE
Vsync trouble

### ISSUE THREAD

**Issue by rennis250** [tomaka/glutin#922]
Hello,

I tried to activate vsync today, but it didn't work on either Mac (10.12.something) or Ubuntu with Wayland. I did nothing more than take a basic example and add ```with_vsync(true)``` to the ContextBuilder. Is this expected behaviour? I also added a section to poll for events, in case that is necessary to force synchronization, but that didn't change anything.

Thanks and best,
Rob

**Comment by rennis250** [comment]
Also tried it on X11 and still no luck. I should have said that on Ubuntu, I am using the Nouveau drivers and have access to OpenGL 4.5 and Mesa 17.0.3

**Comment by mitchmindtree** [comment]
Could you elaborate on what you mean by it not working? Was there an error of some sort? Or does it seem like it's being ignored?

**Comment by rennis250** [comment]
Sorry, yes, I meant that it was being ignored. A window still opened and worked as expected, but it did not sync to the vertical retrace. I see now what I did wrong though... I thought I had added a part for ```poll_events()```, but that was in a different project and I was rather tired. I've now added ```poll_events()``` to the code that uses vsync and it works as expected on the Mac. I'll check on the Ubuntu machine later. Maybe this should be mentioned somewhere in the documentation? I'm not sure if it is true for all OpenGL window/context managers (e.g., GLFW) that one needs to poll events in order to get vsync.

Best,
Rob
